src/Microsoft.Crank.Controller/Ignore/IgnoreFile.cs

```diff
@@ -26,25 +20,22 @@ private IgnoreFile(string gitIgnorePath)
         /// <returns>A list of <see cref="IgnoreRule"/> instances.</returns>
         public static IgnoreFile Parse(string path, bool includeParentDirectories = false)
         {
-            var ignoreFile = new IgnoreFile(path);
+            var ignoreFile = new IgnoreFile();
 
-            var currentDir = new DirectoryInfo(Path.GetDirectoryName(path));
+            var currentDir = Directory.Exists(path) ? new DirectoryInfo(path) : new DirectoryInfo(Path.GetDirectoryName(path));
             
             while (currentDir != null && currentDir.Exists)
             {
-                var gitIgnoreFilename = Path.Combine(currentDir.FullName, ".gitignore");
+                var gitIgnoreFilename = Path.GetFullPath(currentDir.FullName + "/.gitignore");
```

Comment:
Shouldn't this still be a Path.Combine?

Reply:
I preferred `GetGullPath` because 
1- that's what is required, it create the fully qualified path
2- if the second argument starts with `/` it won't root it to the drive

So maybe here it's the same, but I tried to use the same pattern instead of mixing solutions

Reply:
I meant
```
Path.GetFullPath(Path.Combine(currentDir.FullName, ".gitignore"));
```

Reply:
Should work here. I had a unit test failing because of that but maybe it was for another place.